template.yaml

```diff
@@ -0,0 +1,970 @@
+    Properties:
+      # checkov:skip=CKV_AWS_116: DLQ not required
+      # checkov:skip=CKV_AWS_117: Internet access not required
+      ReservedConcurrentExecutions: 1
+      Handler: index.handler
+      MemorySize: 128
+      Runtime: nodejs20.x
+      Timeout: 900
+      InlineCode: |
+        'use strict';
+  
+        const ip4ToInt = ip =>
+          ip.split('.').reduce((int, oct) => (int << 8) + parseInt(oct, 10), 0) >>> 0;
+        
+        const isIp4InCidr = ip => cidr => {
+          const [range, bits = 32] = cidr.split('/');
+          const mask = ~(2 ** (32 - bits) - 1);
+          return mask === 0
+            ? ip4ToInt(ip) === ip4ToInt(range)
+            : (ip4ToInt(ip) & mask) === (ip4ToInt(range) & mask);
+        };
+        
+        const isIp4InCidrs = (ip, cidrs) => cidrs.some(isIp4InCidr(ip));
+        
+        exports.handler = async(event) => {
+          const ipAddress = event.requestContext.http.sourceIp;
+          const validIps = [
+            '217.196.229.77/32',
+            '217.196.229.79/32',
+            '217.196.229.80/31',
+            '51.149.8.0/25',
+            '51.149.8.128/29',
+            '213.86.153.211/32',
+            '213.86.153.212/31',
+            '213.86.153.214/32',
+            '213.86.153.235/32',
+            '213.86.153.236/31',
+            '213.86.153.231/32'];
+            const isValidIp = isIp4InCidrs(ipAddress, validIps);
+            return {
+              'isAuthorized': isValidIp
+            };
+        };
+    Tags:
+      - Key: Name
+        Value: !Sub "${AWS::StackName}-ApiGatewayLambdaFunctionAuthorizer"
+      - Key: Service
+        Value: Relying Party stub
+      - Key: Source
+        Value: alphagov/di-auth-stub-relying-party
+      - Key: Owner
+        Value: di-orchestration@digital.cabinet-office.gov.uk
+
+  ApiGatewayAuthorizer:
+    Type: AWS::ApiGatewayV2::Authorizer
+    Properties:
+      Name: !Sub "${AWS::StackName}-ApiGatewayAuthorizer"
+      ApiId: !Ref ApiGateway
+      AuthorizerResultTtlInSeconds: 60
+      IdentitySource:
+        - $context.identity.sourceIp
+      AuthorizerType: REQUEST
+      AuthorizerPayloadFormatVersion: 2.0
+      EnableSimpleResponses: true
+      AuthorizerUri: !Join
+        - ''
+        - - 'arn:aws:apigateway:'
+          - 'eu-west-2'
+          - ':lambda:path/2015-03-31/functions/'
+          - !GetAtt
+            - ApiGatewayLambdaFunctionAuthorizer
+            - Arn
+          - /invocations
+
+  ApiGatewayAuthorizerInvokePermission:
+    Type: AWS::Lambda::Permission
+    Properties:
+      FunctionName: !GetAtt ApiGatewayLambdaFunctionAuthorizer.Arn
+      Action: "lambda:InvokeFunction"
+      Principal: "apigateway.amazonaws.com"
+      SourceArn: !Join [
+        "", [
+          "arn:aws:execute-api:",
+          !Ref AWS::Region, ":",
+          !Ref AWS::AccountId, ":",
+          !Ref ApiGateway, "/",
+          "authorizers/", !Ref ApiGatewayAuthorizer
+        ]
+      ]
```

Comment:
Is there a reason why this is using the json format instead of the yaml format?

Reply:
Another copy job (this was all from dev platform, I thought they'd be more consistent 😆)